From 9a23bb7bc84039dc65aef7b9feed6d3de0e6b144 Mon Sep 17 00:00:00 2001
From: Martin Pitt <mpitt@redhat.com>
Date: Fri, 31 Mar 2017 16:50:19 +0200
Subject: [PATCH] test: Run with temporary $XDG_CACHE_HOME

Avoid clobbering the real user's home directory with cache files from
the unit tests. This also avoids (fatal) warnings about "couldn't create
runtime dir:" if the user's home directory does not exist (like e. g. in
pbuilder).

Note: We cannot use a temporary $HOME because cockpit-bridge
overwrites it with the value from /etc/passwd; see #6246 for details.

Fixes #6245
Closes #6261
Reviewed-by: Stef Walter <stefw@redhat.com>
---
 src/base1/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/base1/Makefile.am b/src/base1/Makefile.am
index 9251203..f0bd918 100644
--- a/src/base1/Makefile.am
+++ b/src/base1/Makefile.am
@@ -23,7 +23,7 @@ basedebug_DATA = \
 	dist/base1/cockpit.min.js.map \
 	$(NULL)
 
-AM_TESTS_ENVIRONMENT = export G_DEBUG=fatal-criticals,fatal-warnings ;
+AM_TESTS_ENVIRONMENT = export G_DEBUG=fatal-criticals,fatal-warnings ; export XDG_CACHE_HOME=$$(mktemp -d) ;
 
 dist/base1/cockpit.min.css: src/base1/cockpit.css
 	$(AM_V_GEN) $(MKDIR_P) $(dir $@) && \
-- 
2.9.3

